src/services/post.service.js

```diff
@@ -201,5 +271,219 @@ module.exports = {
       err.func = 'getPostsForTimeline'
       throw err
     }
+  },
+
+  /**
+   * Gets data of a single publication, this includes the reference post id
+   * in case that the post is "shared" type.
+   * Includes if the user requesting likes the post.
+   * @param {number} post_id Id of post to get the data.
+   * @param {number} user_id Id of user requesting.
+   * @returns {obejct} An object with:
+   * - id: number,
+   * - username: string,
+   * - firstname: string,
+   * - lastname: string,
+   * - profile_img_src: string,
+   * - content: string,
+   * - img_src: string,
+   * - post_type: string,
+   * - like_counter: number,
+   * - created_at: datetime,
+   * - liked_by_user: bool,
+   * - group_name: string,
+   * - group_id: number,
+   * - referenced_post_id: number
+   */
+  getPostDataUserAuth: async function(post_id, user_id) {
+    const query = `
+      select
+        posts.id,
+        users.username,
+        users.firstname,
+        users.lastname,
+        users.profile_img_src,
+        posts.content,
+        posts.img_src,	
+        posts.post_type,
+        posts.like_counter,
+        posts.created_at,
+        case 
+          when favorite_posts.user_id = ? then 1 
+          else 0
+        end as liked_by_user,
+        case 
+          when posts.post_type = 'group' then (
+            select user_groups.name
+            from posts
+            inner join group_posts
+              on posts.id = group_posts.post_id
+            inner join user_groups
+              on group_posts.group_id = user_groups.id
+            where posts.id = ?
+          )
+        end as group_name,
+        case 
+          when posts.post_type = 'group' then (
+            select user_groups.id
+            from posts
+            inner join group_posts
+              on posts.id = group_posts.post_id
+            inner join user_groups
+              on group_posts.group_id = user_groups.id
+            where posts.id = ?
+          )
+        end as group_id,
+        posts.referenced_post_id
+      from posts
+      inner join users
+        on posts.user_id = users.id
+      left join favorite_posts
+        on posts.id = favorite_posts.post_id
+      where 
+        posts.id = ?
+      limit 1;
+    `
+    try {
+      const post = await mariadb.query(query, [user_id, post_id, post_id, post_id])
+      return post[0]
+    } catch (err) {
+      err.file = __filename
+      err.func = 'getPostDataUserAuth'
+      throw err
+    }
+  },
```

Comment:
This function can work for both, user authenticated and unauthenticated.

Queries for both process are quite similar. For authenticated users I see you just add a few lines more, but you do not afffect the rest of the lines. If you create the query dinamicaly, you can detect if `user_id` is provided (not undefined or not null) and just add the code to query the post for an authenticated user. If `user_id` is missing, do not add the extra lines.

For instance:

```js
const query = `
      select
        posts.id,
        users.username,
        users.firstname,
        users.lastname,
        users.profile_img_src,
        posts.content,
        posts.img_src,	
        posts.post_type,
        posts.like_counter,
        posts.created_at,`
if(user_id) {
  query += `
        case 
          when favorite_posts.user_id = ? then 1 
          else 0
        end as liked_by_user,`
}
query += `
        case 
          when posts.post_type = 'group' then (
            select user_groups.name
            from posts
            inner join group_posts
              on posts.id = group_posts.post_id
            inner join user_groups
              on group_posts.group_id = user_groups.id
            where posts.id = ?
          )
        end as group_name,
        ...`
```
And so on with the rest of the code when you need to use `user_id`.

Use the same approach for:

- `getBasePostDataUserAuth`

Remove *UserAuth* from the name of the functions would be appropriate.